lib/utils.js

```diff
@@ -82,411 +82,450 @@ var truncateMessage = function(parts, max) {
-            if (err) {
-                return callback(err);
-            }
-            callback(null, data.rows);
-        });
-    },
-    getForm: function(form_code) {
-        var forms = config.get('forms');
-        return forms && forms[form_code];
-    },
-    isFormCodeSame: function(form_code, test) {
-        // case insensitive match with junk padding
-        return (new RegExp('^\W*' + form_code + '\\W*$','i')).test(test);
-    },
+        if (!results.rows.length) {
+          return callback();
+        }
 
-    /*
-     * Return message from configured translations given key and locale.
-     *
-     * If translation is not found return the translation key.  Otherwise
-     * messages won't get added because of an empty message.  Better to at
-     * least surface something in the UI providing a clue that something is
-     * misconfigured as opposed to broken.
-     *
-     * @param {String} key - translation key/identifier
-     * @param {String} locale - short locale string
-     *
-     * @returns {String|undefined} - the translated message
-     */
-    translate: function(key, locale) {
-        var translations = config.getTranslations();
-        var msg = (translations[locale] && translations[locale][key]) ||
-                  (translations.en && translations.en[key]) ||
-                  key;
-        return msg.trim();
-    },
-    escapeRegex: function(s) {
-        return s.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&');
-    },
-    /*
-     * Return false when the recipient phone matches the denied list.
-     *
-     * outgoing_deny_list is a comma separated list of strings. If a string in
-     * that list matches the beginning of the phone then we set up a response
-     * with a denied state. The pending message process will ignore these
-     * messages and those reports will be left without an auto-reply. The
-     * denied messages still show up in the messages export.
-     *
-     * @param {String} from - Recipient phone number
-     * @returns {Boolean}
-     */
-    isOutgoingAllowed: function(from) {
-        var self = module.exports,
-            conf = config.get('outgoing_deny_list') || '';
-        if (!from) {
-            return true;
+        if (results.rows.length > 1) {
+          console.warn('More than one patient person document for shortcode ' + patientShortcodeId);
         }
-        if (self._isMessageFromGateway(from)) {
-            return false;
+
+        return callback(null, results.rows[0].id);
+    });
+  },
+  /*
+   * Used to avoid infinite loops of auto-reply messages between gateway and
+   * itself.
+   */
+  _isMessageFromGateway: from => {
+    const gw = config.get('gateway_number');
+    if (typeof gw === 'string' && typeof from === 'string') {
+      return phoneUtil.isNumberMatch(gw, from) >= 3;
+    }
+    return false;
+  },
+  hydrateDoc: (db, id, callback) => {
+    db.medic.view('medic-client', 'docs_by_id_lineage', {
```

Comment:
Hmm maybe. We use words after "by" to indicate it's part of the key, and the lineage depth is part of the key. In the view definition it _looks_ like it's the value, but what you get back is just a list of docs. Let's talk about this one...